src/h5web/providers/hdf5-models.ts

```diff
@@ -12,37 +12,6 @@ export enum HDF5Collection {
   Datatypes = 'datatypes',
 }
 
-export interface HDF5Metadata {
-  root: HDF5Id;
-  [HDF5Collection.Groups]: Record<HDF5Id, HDF5Group>;
-  [HDF5Collection.Datasets]?: Record<HDF5Id, HDF5Dataset>;
-  [HDF5Collection.Datatypes]?: Record<HDF5Id, HDF5Datatype>;
-}
-
-/* -------------------- */
-/* ----- ENTITIES ----- */
-
-export interface HDF5Group {
-  alias: [string];
-  attributes?: HDF5Attribute[];
-  links?: HDF5Link[];
-}
-
-export interface HDF5Dataset<
-  S extends HDF5Shape = HDF5Shape,
-  T extends HDF5Type = HDF5Type
-> {
-  alias: [string];
-  attributes?: HDF5Attribute[];
-  shape: S;
-  type: T;
-}
-
-export interface HDF5Datatype<T = HDF5Type> {
-  alias: [string];
-  type: T;
-}
```

Comment:
I move all these now-HSDS-specific models to `providers/hsds/models`. I also rename them with the `Hsds` prefix and remove the generics. We can now see them almost as the response models of HSDS' endpoints (it will be the case even more after the second wave of changes).